comp/otelcol/configstore/impl/configstore.go

```diff
@@ -0,0 +1,99 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package configstoreimpl provides the implementation of the otel-agent configstore.
+package configstoreimpl
+
+import (
+	"sync"
+
+	configstore "github.com/DataDog/datadog-agent/comp/otelcol/configstore/def"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/otelcol"
+	"gopkg.in/yaml.v2"
+)
+
+type configStoreImpl struct {
+	provided *otelcol.Config
+	enhanced *otelcol.Config
+	mu       sync.RWMutex
+}
```

Comment:
Can we add unit tests for this ?

Reply:
What type of unit tests did you have in mind ? These tests: https://github.com/DataDog/datadog-agent/blob/d4926ceddcbe42a8f8568a5f87385b5999010a4a/comp/otelcol/collector/impl/collector_test.go should cover the configstore logic ?